src/main/java/org/thoughtslive/jenkins/plugins/jira/steps/AddCommentStep.java

```diff
@@ -88,14 +106,21 @@ protected Execution(final AddCommentStep step, final StepContext context)
 
       if (response == null) {
         final String idOrKey = Util.fixEmpty(step.getIdOrKey());
-        final String comment = Util.fixEmpty(step.getComment());
+
+        if (step.getComment() != null && step.getInput() != null) {
+          errorMessage = "Use comment or either input.";
+        }
+
+        if (step.getComment() == null && step.getInput() == null) {
+          errorMessage = "You need to set at least comment or input.";
+        }
 
         if (idOrKey == null) {
```

Comment:
Lets add couple of util methods in `InputBuilder` class to verify `body` is not null or empty and also to add the metadata (is to maintain the passivity with `auditLog`)

```java
static getField(input, fieldName) {
    return input."$fieldName"
}

static setField(input, fieldName, fieldvalue) {
    input."$fieldName" = fieldvalue
}
```

Get the body (comment) field and also set the metadata by:

```java
String message = InputBuilder.getField(step.input, "body") == null ? InputBuilder.getField(step.input, "body").toString(): "";
InputBuilder.setField(step.input, "body", addMeta(message));
```